rllib/policy/tf_policy.py

```diff
@@ -89,7 +92,15 @@ def __init__(self,
                 placeholders during loss computation.
             model (rllib.models.Model): used to integrate custom losses and
                 stats from user-defined RLlib models.
-            action_logp (Tensor): log probability of the sampled action.
+            sampled_action_logp (Tensor): log probability of the sampled
+                action.
+            action_input (Optional[Tensor]): Input placeholder for actions for
+                logp/log-likelihood calculations.
+            action_dist_class (Optional[class]): An optional action
+                distribution class to be used for aciton log-likelihood
+                calculations.
+            action_dist_parameters (Optional[Tensor]): An optional placeholder
```

Comment:
It's a bit confusing to call this "parameters", since model variables are also called this. Some other term like dist_input_tensor?

Actually another question, why not instead pass in `likelihood_fn(obs, action)`? So there will be two cases:
1. PPO/PG, etc.: can be computed automatically (don't need to pass in likelihood_fn)
2. SAC/DQN define their own likelihood_fn().

I think this strategy would be cleaner since you only have to pass in one argument (likelihood fn).

Reply:
Ok, will change.

Reply:
Reduced action_dist_class, action_dist_parameters to sigle log_likelihood op, which is passed in as None or generated by the DynamicTFPolicy or by the custom TFPolicy child-class.

Reply:
Taking out the SAC is very hard to do here (which is the reason, I moved it into this PR; I know it's not ideal).
I can revert all the changes in the SAC itself, but then I will have to add some code to it to make it work with the new log-likelihood estimator (which requires a distribution class and distribution inputs, which SAC currently does NOT provide). So this means I will also have to leave the new SquashedGaussian in here.

Reply:
Removed SAC changes and added the extra log_likelihood_fn to it. Should work now. Waiting for tests ...